src/frontend/src/expr/table_function.rs

```diff
@@ -150,13 +150,19 @@ impl TableFunction {
             {
                 let (bucket, _) = extract_bucket_and_file_name(&eval_args[5].clone())?;
 
-                let is_minio = eval_args[2].starts_with("http");
+                let (s3_region, s3_endpoint) = match eval_args[2].starts_with("http") {
+                    true => ("us-east-1".to_owned(), eval_args[2].clone()), /* for minio, hard code region as not used but needed. */
+                    false => (
+                        eval_args[2].clone(),
+                        format!("https://{}.s3.{}.amazonaws.com", bucket, eval_args[2],),
```

Comment:
Should we make the endpoint optional?

Reply:
I suggest no, since it goes directly as a parameter to the new s3_operator.